scrapy/responsetypes.py

```diff
@@ -136,6 +148,12 @@ def from_args(self, headers=None, url=None, filename=None, body=None):
 
         if headers and b'Content-Type' in headers:
             content_types = tuple(headers.getlist(b'Content-Type'))
+        elif headers and b'Content-Disposition' in headers:
+            content_types = (self.__guess_content_type(content_disposition=headers.get(b'Content-Disposition')).encode(),)
+        elif url:
+            content_types = (self.__guess_content_type(url=url).encode(),)
+        elif filename:
+            content_types = (self.__guess_content_type(filename=filename).encode(),)
```

Comment:
I suggest you modify the `__guess_content_type` function so that here you can replace all these lines with a single call to that function.

Reply:
What should be the priority of these parameters content-disposition, URL , filename?

Reply:
Yes, to keep the same priorities as the earlier implementation. You could also move the Content-Type part into the function.